cvmfs/options.h

```diff
@@ -15,24 +15,158 @@
+   *                     repository. If false the configuration file is in /etc
+   */
+  virtual void ParsePath(const std::string &config_file,
+      const bool external) = 0;
+
+  /**
+   * Parses the default config files for cvmfs.
+   */
+  void ParseDefault(const std::string &fqrn);
+
+  /**
+   * Cleans all information about the variables.
+   */
+  void ClearConfig();
+
+  /**
+   * Checks if a concrete key (variable) is defined
+   *
+   * @param   key variable to be checked in the map
+   * @return  true if there is a value for key, false otherwise.
+   */
+  bool IsDefined(const std::string &key);
+
+  /**
+   * Gets the stored value for a concrete variable
+   *
+   * @param  key variable to be accessed in the map
+   * @param  value container of the received value, if it exists
+   * @return true if there was a value stored in the map for key
+   */
+  bool GetValue(const std::string &key, std::string *value);
+
+  /**
+   * Gets the stored last source of a concrete variable
+   *
+   * @param  key variable to be accessed in the map
+   * @param  value container of the received value, if it exists
+   * @return true if that variable was previously defined in
+   *         at least one source
+   */
+  bool GetSource(const std::string &key, std::string *value);
+
+  /**
+   * Checks if a variable contains a boolean value
+   *
+   * @param   param_value variable to be accessed in the map
+   * @return  true if param has as value "YES", "ON" or "1". False otherwise.
+   */
+  bool IsOn(const std::string &param_value);
+
+  /**
+   * Retrieves a vector containing all stored keys.
+   *
+   * @return a vector with all keys contained in the map
+   */
+  std::vector<std::string> GetAllKeys();
 
-void Init();
-void Fini();
+  /**
+   * Gets all stored key-values of the map in an string format. This format
+   * follows the following pattern:
+   *
+   * "KEY=VALUE    # from SOURCE"
+   *
+   * @return a vector containing all key-values in a string format.
+   */
+  std::string Dump();
 
-void ParsePath(const std::string &config_file, const bool external);
-void ParseDefault(const std::string &fqrn);
-void ClearConfig();
-bool IsDefined(const std::string &key);
-bool GetValue(const std::string &key, std::string *value);
-bool GetSource(const std::string &key, std::string *value);
-bool IsOn(const std::string &param_value);
-std::vector<std::string> GetAllKeys();
-std::string Dump();
+  /**
+   * Parse a configuration file whose variables's values are positive integers
+   *
+   * @param  path absolute path to the configuration file
+   * @param  map map to store the generated key-value
+   * @return true if the file exists, is readable and all values can be parsed
+   *  to integers
+   */
+  bool ParseUIntMap(const std::string &path, std::map<uint64_t, uint64_t> *map);
+
+ protected:
+  bool HasConfigRepository(const std::string &fqrn, std::string *config_path);
+
+ protected:
```

Comment:
I told José to do it like that. Usually I follow this convention:

```C++
class Foo {
 public:
  <<< public internal data structures, enums and typedefs  >>>
 private:
  <<< private internal data structures, enums and typedefs >>>

 public:
  <<< public methods >>>

 protected:
  <<< protected methods >>>

 protected:
  <<< protected class members >>>

 private:
  <<< private class members >>>
}
```

Hence, the duplicated `protected` keyword separates two semantic blocks of the class.